main.py

```diff
@@ -159,30 +164,28 @@ def song_format(
     Returns:
         A string presenting the given song information in a human-readable way.
     """
-    # a valid tag is string with at least one non-whitespace character
-    def is_valid_tag(tag: Optional[str]) -> bool:
-        return tag is not None and tag.strip()
-
     content = ""
-    tags = None
 
     # load tags
     try:
-        tags = TinyTag.get(local_filepath)
-    except TinyTagException:
-        # Ignore and move on
-        pass
+        tags = MutagenFile(local_filepath)
+        if isinstance(tags, ID3FileType):
+            # ID3 keys are complicated. EasyID3 unifies the interface with other types
+            tags = EasyID3(local_filepath)
+    except MutagenError:
+        # Ignore file and move on
+        tags = None
 
     # Display in the format <Artist-tag> - <Title-tag>
     # If no artist tag use fallback if valid. Otherwise, skip artist
-    if tags and is_valid_tag(tags.artist):
-        content += tags.artist + " - "
-    elif is_valid_tag(artist_fallback):
+    if tags and "artist" in tags and len(tags["artist"]) > 0:
+        content += str(tags["artist"][0]) + " - "
+    elif artist_fallback is not None:
         content += artist_fallback + " - "
 
     # Always display either title or beautified filename
-    if tags and is_valid_tag(tags.title):
-        content += tags.title
+    if tags and "title" in tags and len(tags["title"]) > 0:
+        content += tags["title"][0]
```

Comment:
Wouldn't this fail if `tags["artist"] == []` ? It seems strange that it could be the case that either the artist field doesn't exist OR the list it maps to is empty, but I can't tell which one happens (or both) from the documentation. I'm not at a computer to check rn.

Reply:
Good question! In my tests, the `_getitem_` method of `EasyID3` will only ever return a list with some items, or `None`. Poking about in the library, that [seems to be the intention](https://github.com/quodlibet/mutagen/blob/30f373fa6d56e1afa17d48a0c6f3e111267fbd32/mutagen/easyid3.py#L76).

```
>>> type(easy_ghost)
<class 'mutagen.easyid3.EasyID3'>

>>> easy_ghost["album"]
Traceback (most recent call last):
  File "/usr/lib/python3.10/code.py", line 90, in runcode
    exec(code, self.locals)
  File "<input>", line 1, in <module>
  File "/home/user/code/busty/env/lib/python3.10/site-packages/mutagen/easyid3.py", line 213, in __getitem__
    return func(self.__id3, key)
  File "/home/user/code/busty/env/lib/python3.10/site-packages/mutagen/easyid3.py", line 120, in getter
    return list(id3[frameid])
  File "/home/user/code/busty/env/lib/python3.10/site-packages/mutagen/_util.py", line 537, in __getitem__
    return self.__dict[key]
KeyError: 'TALB'

>>> easy_ghost.get("album")

>>> easy_ghost.get("album", [])
[]
```